// @quarkly/components/build/cjs/prod/formspree.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _atomize = _interopRequireDefault(require("@quarkly/atomize"));

var _widgets = require("@quarkly/widgets");

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n\tmin-height: 32px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n\tpadding: 16px 32px;\n\tborder: 2px solid;\n\tmin-width: 100px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var NoEndPoint = _atomize["default"].div(_templateObject());

var Wrapper = _atomize["default"].div(_templateObject2());

var url = 'https://formspree.io/';

var Formspree = function Formspree(props) {
  var children = props.children,
      _props$endpoint = props.endpoint,
      endpoint = _props$endpoint === void 0 ? '' : _props$endpoint,
      errorMessage = props.errorMessage,
      completeText = props.completeText,
      rest = (0, _objectWithoutProperties2["default"])(props, ["children", "endpoint", "errorMessage", "completeText"]);

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      complete = _useState2[0],
      setComplete = _useState2[1];

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      fetching = _useState4[0],
      setFetching = _useState4[1];

  var _useState5 = (0, _react.useState)(false),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      error = _useState6[0],
      setError = _useState6[1];

  var action = url + endpoint.trim().replace(url, '');

  var onSubmit = /*#__PURE__*/function () {
    var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(event) {
      var form, data, button, request;
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              if (!fetching) {
                _context.next = 2;
                break;
              }

              return _context.abrupt("return");

            case 2:
              event.preventDefault();
              setError(false);
              setFetching(true);
              form = event.target;
              data = new FormData(form);
              button = form.querySelector('[type="submit"]');
              button && button.setAttribute('disabled', true);
              request = new XMLHttpRequest();
              request.open('POST', action);
              request.setRequestHeader('Accept', 'application/json');

              request.onreadystatechange = function () {
                if (request.readyState !== XMLHttpRequest.DONE) return;

                if (request.status === 200) {
                  setComplete(true);
                } else {
                  setError(true);
                }
              };

              request.send(data);
              setFetching(false);
              button && button.removeAttribute('disabled');

            case 16:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));

    return function onSubmit(_x) {
      return _ref.apply(this, arguments);
    };
  }();

  if (!endpoint || typeof endpoint !== 'string') {
    return /*#__PURE__*/_react["default"].createElement(Wrapper, rest, /*#__PURE__*/_react["default"].createElement(NoEndPoint, null, "Create a form on", ' ', /*#__PURE__*/_react["default"].createElement(_widgets.Link, {
      color: "currentColor",
      "text-decoration": "underline",
      target: "_blank",
      rel: "noopener noreferrer",
      href: "https://formspree.io/"
    }, "formspree.io"), ' ', "and fill in the ", /*#__PURE__*/_react["default"].createElement(_widgets.Strong, null, "endpoint"), " field on the ", /*#__PURE__*/_react["default"].createElement(_widgets.Strong, null, "props panel")));
  }

  if (!complete) {
    return /*#__PURE__*/_react["default"].createElement(Wrapper, rest, /*#__PURE__*/_react["default"].createElement("form", {
      encType: "multipart/form-data",
      method: "post",
      onSubmit: onSubmit,
      action: action
    }, children, error ? /*#__PURE__*/_react["default"].createElement(_widgets.Text, {
      font: "--primary",
      color: "--red"
    }, errorMessage) : null));
  }

  return /*#__PURE__*/_react["default"].createElement(Wrapper, rest, completeText);
};

var _default = (0, _atomize["default"])(Formspree)({
  name: 'Formspree',
  description: {
    en: 'Formspree — supercharge your forms'
  },
  propInfo: {
    endpoint: {
      weight: 1,
      category: 'Main',
      control: 'input'
    },
    completeText: {
      weight: 1,
      category: 'Main',
      control: 'input'
    },
    errorMessage: {
      weight: 1,
      category: 'Main',
      control: 'input'
    }
  }
}, {
  endpoint: '',
  errorMessage: 'Something went wrong',
  completeText: 'Success'
});

exports["default"] = _default;
