// @quarkly/components/build/cjs/prod/googlemap.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _atomize = _interopRequireDefault(require("@quarkly/atomize"));

var Map = function Map(_ref) {
  var _ref$query = _ref.query,
      query = _ref$query === void 0 ? 'New York' : _ref$query,
      _ref$apiKey = _ref.apiKey,
      apiKey = _ref$apiKey === void 0 ? 'AIzaSyA1TA1w9GrLZNaSfAZDaW0lfvGOiL7KULc' : _ref$apiKey,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["query", "apiKey"]);
  return /*#__PURE__*/_react["default"].createElement("div", props, /*#__PURE__*/_react["default"].createElement("iframe", {
    style: {
      position: 'absolute',
      width: '100%',
      height: '100%',
      top: 0,
      left: 0
    },
    frameBorder: "0",
    allowFullScreen: true,
    src: "https://www.google.com/maps/embed/v1/place?key=".concat(apiKey, "&q=").concat(query)
  }));
};

var GoogleMap = (0, _atomize["default"])(Map)({
  name: 'GoogleMap',
  propInfo: {
    apiKey: {
      control: 'input',
      category: 'Main'
    },
    query: {
      control: 'input',
      category: 'Main'
    }
  }
}, {
  width: '100%',
  height: '450px',
  position: 'relative'
});
var _default = GoogleMap;
exports["default"] = _default;
