// @quarkly/components/build/cjs/prod/sectionContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = SectionContent;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _widgets = require("@quarkly/widgets");

function SectionContent(props) {
  var iw = props.iw,
      imaxw = props.imaxw,
      iminw = props.iminw,
      children = props.children,
      restProps = (0, _objectWithoutProperties2["default"])(props, ["iw", "imaxw", "iminw", "children"]);
  return /*#__PURE__*/_react["default"].createElement(_widgets.Box, (0, _extends2["default"])({
    display: "flex",
    "flex-direction": "column",
    width: iw,
    "max-width": imaxw,
    "min-width": iminw,
    "min-height": "16px",
    margin: "0 auto"
  }, restProps, {
    "data-qkind": "SectionContent"
  }), children);
}

Object.assign(SectionContent, {
  effects: {
    hover: ':hover'
  }
});
