// @quarkly/components/build/cjs/prod/socialMedia.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _widgets = require("@quarkly/widgets");

var _hooks = require("../hooks");

var _fa = require("react-icons/fa");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var social = [{
  icon: _fa.FaFacebook,
  propName: 'facebook'
}, {
  icon: _fa.FaWhatsapp,
  propName: 'whatsapp'
}, {
  icon: _fa.FaTumblr,
  propName: 'tumblr'
}, {
  icon: _fa.FaInstagram,
  propName: 'instagram'
}, {
  icon: _fa.FaTwitter,
  propName: 'twitter'
}, {
  icon: _fa.FaGooglePlus,
  propName: 'google-plus'
}, {
  icon: _fa.FaViber,
  propName: 'viber'
}, {
  icon: _fa.FaSnapchat,
  propName: 'snapchat'
}, {
  icon: _fa.FaVk,
  propName: 'vkontakte'
}, {
  icon: _fa.FaPinterest,
  propName: 'pinterest'
}, {
  icon: _fa.FaLinkedinIn,
  propName: 'linked-in'
}, {
  icon: _fa.FaTelegram,
  propName: 'telegram'
}, {
  icon: _fa.FaReddit,
  propName: 'reddit'
}, {
  icon: _fa.FaYoutube,
  propName: 'youtube'
}, {
  icon: _fa.FaFlickr,
  propName: 'flikr'
}, {
  icon: _fa.FaFoursquare,
  propName: 'foursquare'
}, {
  icon: _fa.FaDiscord,
  propName: 'discord'
}, {
  icon: _fa.FaGithubAlt,
  propName: 'github'
}];
var socialProps = social.reduce(function (acc, _ref) {
  var propName = _ref.propName;
  return _objectSpread({}, acc, (0, _defineProperty2["default"])({}, propName, true));
}, {});

var filterProps = function filterProps(props) {
  var result = {};

  for (var propName in props) {
    if (!socialProps[propName]) {
      result[propName] = props[propName];
    }
  }

  return result;
};

var defaultProps = {
  display: 'flex',
  'icon-size': '24px',
  'align-items': 'center',
  'justify-content': 'center'
};
var overrides = {
  icon: {
    props: {}
  },
  link: {
    kind: 'Link',
    props: {
      padding: '8px',
      margin: '0 4px',
      color: '--primary',
      background: '--color-lightD1',
      'hover-background': '--color-lightD2',
      'border-radius': '4px'
    }
  }
};

var SocialMedia = function SocialMedia(_ref2) {
  var size = _ref2['icon-size'],
      props = (0, _objectWithoutProperties2["default"])(_ref2, ["icon-size"]);

  var _useOverrides = (0, _hooks.useOverrides)(filterProps(props), overrides, defaultProps),
      override = _useOverrides.override,
      rest = _useOverrides.rest;

  return /*#__PURE__*/_react["default"].createElement(_widgets.Box, rest, social.map(function (_ref3) {
    var propName = _ref3.propName,
        icon = _ref3.icon;
    if (!props[propName]) return null;
    return /*#__PURE__*/_react["default"].createElement(_widgets.Link, (0, _extends2["default"])({
      key: propName,
      href: props[propName],
      target: "_blank"
    }, override('link', "link-".concat(propName))), /*#__PURE__*/_react["default"].createElement(_widgets.Icon, (0, _extends2["default"])({
      icon: icon,
      size: size
    }, override('icon', "icon-".concat(propName)))));
  }));
};

var propInfo = {
  'icon-size': {
    control: 'input',
    transformer: 'pixel',
    category: 'Main'
  }
};
social.map(function (_ref4) {
  var propName = _ref4.propName;
  propInfo[propName] = {
    weight: 1,
    control: 'input',
    category: 'Social Links'
  };
});
Object.assign(SocialMedia, {
  defaultProps: defaultProps,
  overrides: overrides,
  propInfo: propInfo
});
var _default = SocialMedia;
exports["default"] = _default;
