// @quarkly/components/build/cjs/prod/stackItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _atomize = _interopRequireDefault(require("@quarkly/atomize"));

var _widgets = require("@quarkly/widgets");

var _stackItemContent = _interopRequireDefault(require("./stackItemContent"));

var _hooks = require("../hooks");

var overrides = {
  StackItemContent: {
    kind: 'StackItemContent',
    props: {}
  }
};

var StackItem = function StackItem(_ref) {
  var gap = _ref.gap,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["gap"]);

  var _useOverrides = (0, _hooks.useOverrides)(props, overrides, {}),
      override = _useOverrides.override,
      children = _useOverrides.children,
      rest = _useOverrides.rest;

  return /*#__PURE__*/_react["default"].createElement(_widgets.Box, (0, _extends2["default"])({}, rest, {
    "box-sizing": "border-box",
    padding: "calc(".concat(gap, " / 2)")
  }), /*#__PURE__*/_react["default"].createElement(_stackItemContent["default"], override('StackItemContent'), children));
};

StackItem.effects = {
  hover: ':hover'
};

var _default = Object.assign(StackItem, {
  overrides: overrides
});

exports["default"] = _default;
