// @quarkly/components/build/cjs/prod/stack.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _widgets = require("@quarkly/widgets");

var Stack = function Stack(_ref) {
  var gap = _ref.gap,
      children = _ref.children,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["gap", "children"]);
  return /*#__PURE__*/_react["default"].createElement(_widgets.Box, (0, _extends2["default"])({
    margin: "calc(".concat(gap, " * -0.5)")
  }, rest), _react["default"].Children.map(children, function (child) {
    return _react["default"].isValidElement(child) ? _react["default"].cloneElement(child, {
      gap: gap
    }) : child;
  }));
};

Stack.propInfo = {
  gap: {
    weight: '1',
    control: 'input',
    library: 'components.stack.gap',
    transformer: 'pixel',
    category: 'Main'
  }
};
Stack.defaultProps = {
  gap: '--cmp-stack-gap-default',
  display: 'flex',
  'flex-wrap': 'wrap'
};
Stack.effects = {
  hover: ':hover'
};
var _default = Stack;
exports["default"] = _default;
