// @quarkly/widgets/build/cjs/utils/abstractLink.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AbstractLink = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _reactRouterHashLink = require("react-router-hash-link");

var _gatsbyLink = _interopRequireDefault(require("gatsby-link"));

var GatsbyHashLink = function GatsbyHashLink(props) {
  return (0, _reactRouterHashLink.genericHashLink)(props, _gatsbyLink["default"]);
};

var getQAPI = function getQAPI() {
  var globalThis = null;

  if (typeof window !== 'undefined') {
    globalThis = window;
  } else if (typeof global !== 'undefined') {
    globalThis = global;
  } else {
    return {};
  }

  if (!globalThis.QAPI) {
    return {};
  }

  return globalThis.QAPI;
};

var isAbsoluteURL = function isAbsoluteURL(url) {
  if (typeof url !== 'string') {
    return false;
  } // Don't match Windows paths `c:\`


  if (/^[a-zA-Z]:\\/.test(url)) {
    return false;
  } // Scheme: https://tools.ietf.org/html/rfc3986#section-3.1
  // Absolute URL: https://tools.ietf.org/html/rfc3986#section-4.3


  return /^[a-zA-Z][a-zA-Z\d+\-.]*:/.test(url);
};

var AbstractLink = (0, _react.forwardRef)(function (props, ref) {
  var _props$href = props.href,
      href = _props$href === void 0 ? '' : _props$href,
      children = props.children,
      rest = (0, _objectWithoutProperties2["default"])(props, ["href", "children"]);

  var _getQAPI = getQAPI(),
      projectType = _getQAPI.projectType,
      mode = _getQAPI.mode;

  var Link = projectType === 'gatsby' ? GatsbyHashLink : _reactRouterHashLink.HashLink;

  if (mode !== 'development' && isAbsoluteURL(href)) {
    return _react["default"].createElement("a", (0, _extends2["default"])({
      href: href,
      ref: ref
    }, rest), children);
  }

  return _react["default"].createElement(Link, (0, _extends2["default"])({
    to: href,
    ref: ref
  }, rest), children);
});
exports.AbstractLink = AbstractLink;
