// @quarkly/widgets/build/esm/prod/index.js
export { default as Theme, Provider } from './theme';
export { default as Image } from './image';
export { default as Button } from './button';
export { default as Text } from './text';
export { default as Hr } from './hr';
export { default as Box } from './box';
export { default as List } from './list';
export { default as Select } from './select';
export { default as Option } from './option';
export { default as Input } from './input';
export { default as Inline } from './inline';
export { default as Icon } from './icon';
export { default as Section } from './section';
export { default as SectionContent } from './sectionContent';
export { default as Structure } from './structure';
export { default as StructureContent } from './structureContent';
export { default as Placeholder } from './placeholder'; // инлайновые элементы визивига

export { Em, I } from './em';
export { Strong, B } from './strong';
export { Strike, S } from './strike';
export { default as Link } from './link';
export { default as LinkBox } from './linkBox';
export { default as Span } from './span';
export { default as U } from './u';
