// @quarkly/widgets/build/cjs/prod/placeholder.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _index = require("./index");

var Placeholder = (0, _react.forwardRef)(function (_ref, ref) {
  var message = _ref.message,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["message"]);
  return _react["default"].createElement(_index.Box, (0, _extends2["default"])({
    ref: ref,
    height: "100%",
    "min-height": "64px",
    background: "rgba(228, 232, 236, 0.2)",
    border: "2px dashed --color-lightD2",
    "align-items": "center",
    "justify-content": "center",
    display: "flex"
  }, props), _react["default"].createElement(_index.Text, {
    "font-family": "--sans",
    "font-size": "16px",
    "line-height": "1.2",
    "font-weight": "300",
    "letter-spacing": "1.5px",
    "text-align": "center",
    color: "palegrey",
    opacity: "0.8"
  }, message));
});
var _default = Placeholder;
exports["default"] = _default;
