// @quarkly/widgets/build/cjs/prod/section.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _box = _interopRequireDefault(require("./box"));

var _sectionContent = _interopRequireDefault(require("./sectionContent"));

var _components = require("@quarkly/components");

var _commonProps = require("./common-props");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var overrides = {
  sectionContent: {
    props: {}
  }
};

var defaultProps = _objectSpread({
  as: 'section',
  width: '100%',
  margin: '0',
  padding: '24px 0',
  'inner-width': '--cmp-section-width-default',
  'inner-min-width': '--cmp-section-minWidth-default',
  'inner-max-width': '--cmp-section-maxWidth-default',
  'lazy-load': true
}, _commonProps.pointerEventsCategoryBaseDefaults);

var Section = (0, _react.forwardRef)(function (_ref, forwardedRef) {
  var iw = _ref['inner-width'],
      imaxw = _ref['inner-max-width'],
      iminw = _ref['inner-min-width'],
      lazyLoad = _ref['lazy-load'],
      props = (0, _objectWithoutProperties2["default"])(_ref, ["inner-width", "inner-max-width", "inner-min-width", "lazy-load"]);

  var _useOverrides = (0, _components.useOverrides)(props, overrides, defaultProps),
      override = _useOverrides.override,
      children = _useOverrides.children,
      rest = _useOverrides.rest;

  var ref = (0, _react.useRef)(null);
  (0, _react.useImperativeHandle)(forwardedRef, function () {
    return ref.current;
  });
  var render = (0, _components.useLazyHydrate)({
    strategy: _components.WHEN_VISIBLE
  }, ref);

  if (!render && lazyLoad) {
    return _react["default"].createElement(_box["default"], (0, _extends2["default"])({
      display: "flex"
    }, rest, {
      ref: ref,
      dangerouslySetInnerHTML: {
        __html: ''
      }
    }));
  }

  return _react["default"].createElement(_box["default"], (0, _extends2["default"])({
    display: "flex"
  }, rest, {
    ref: ref
  }), _react["default"].createElement(_sectionContent["default"], (0, _extends2["default"])({
    iw: iw,
    imaxw: imaxw,
    iminw: iminw
  }, override('SectionContent')), children));
});

var propInfo = _objectSpread({
  'inner-width': {
    weight: 1,
    library: 'components.section.width',
    control: 'input',
    transformer: 'width',
    category: 'Main'
  },
  'inner-min-width': {
    weight: 1,
    library: 'components.section.minWidth',
    control: 'input',
    transformer: 'width',
    category: 'Main'
  },
  'inner-max-width': {
    weight: 1,
    library: 'components.section.maxWidth',
    control: 'input',
    transformer: 'width',
    category: 'Main'
  },
  'lazy-load': {
    type: 'boolean',
    control: 'checkbox',
    category: 'Main'
  }
}, _commonProps.commonProps, {}, _commonProps.pointerEventsCategoryBase);

var effects = {
  hover: ':hover'
};

var _default = Object.assign(Section, {
  overrides: overrides,
  defaultProps: defaultProps,
  effects: effects,
  propInfo: propInfo
});

exports["default"] = _default;
