// @quarkly/widgets/build/cjs/prod/structure.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _index = require("./index");

var _components = require("@quarkly/components");

var _commonProps = require("./common-props");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var overrides = {
  Content: {
    props: {}
  }
};
var Structure = (0, _react.forwardRef)(function (_ref, forwardedRef) {
  var iw = _ref['inner-width'],
      imaxw = _ref['inner-max-width'],
      iminw = _ref['inner-min-width'],
      lazyLoad = _ref['lazy-load'],
      cellsNumberTotal = _ref['cells-number-total'],
      cellsNumberGroup = _ref['cells-number-group'],
      props = (0, _objectWithoutProperties2["default"])(_ref, ["inner-width", "inner-max-width", "inner-min-width", "lazy-load", "cells-number-total", "cells-number-group"]);

  var _useOverrides = (0, _components.useOverrides)(props, overrides),
      override = _useOverrides.override,
      children = _useOverrides.children,
      rest = _useOverrides.rest;

  var cellsTotal = (0, _react.useMemo)(function () {
    return parseInt(cellsNumberTotal) > 0 ? parseInt(cellsNumberTotal) : 1;
  }, [cellsNumberTotal]);
  var cellsGroup = (0, _react.useMemo)(function () {
    return parseInt(cellsNumberGroup) > 0 ? parseInt(cellsNumberGroup) : 1;
  }, [cellsNumberGroup]);
  var ref = (0, _react.useRef)(null);
  (0, _react.useImperativeHandle)(forwardedRef, function () {
    return ref.current;
  });
  var render = (0, _components.useLazyHydrate)({
    strategy: _components.WHEN_VISIBLE
  }, ref);

  if (!render && lazyLoad) {
    return _react["default"].createElement(_index.Box, (0, _extends2["default"])({
      display: "flex"
    }, rest, {
      ref: ref,
      dangerouslySetInnerHTML: {
        __html: ''
      }
    }));
  }

  return _react["default"].createElement(_index.Box, (0, _extends2["default"])({
    margin: "0",
    padding: "24px 0",
    width: "100%",
    display: "flex"
  }, rest, {
    ref: ref
  }), _react["default"].createElement(_index.StructureContent, (0, _extends2["default"])({
    iw: iw,
    imaxw: imaxw,
    iminw: iminw,
    cellsTotal: cellsTotal,
    cellsGroup: cellsGroup
  }, override('Content'))), children);
});

var propInfo = _objectSpread({
  'cells-number-total': {
    title: 'Total number of cells',
    control: 'input',
    type: 'number',
    category: 'Cells',
    weight: 1
  },
  'cells-number-group': {
    title: 'Number of cells in the pattern',
    control: 'input',
    type: 'number',
    category: 'Cells',
    weight: 1
  },
  'inner-width': {
    title: 'Content width',
    control: 'input',
    library: 'components.section.width',
    transformer: 'width',
    category: 'Main',
    weight: 1
  },
  'inner-min-width': {
    title: 'Content min width',
    control: 'input',
    library: 'components.section.minWidth',
    transformer: 'width',
    category: 'Main',
    weight: 1
  },
  'inner-max-width': {
    title: 'Content max width',
    control: 'input',
    library: 'components.section.maxWidth',
    transformer: 'width',
    category: 'Main',
    weight: 1
  },
  'lazy-load': {
    type: 'boolean',
    control: 'checkbox',
    category: 'Main'
  }
}, _commonProps.commonProps, {}, _commonProps.pointerEventsCategoryBase);

var defaultProps = _objectSpread({
  'cells-number-total': '3',
  'inner-width': '--cmp-section-width-default',
  'inner-min-width': '--cmp-section-minWidth-default',
  'inner-max-width': '--cmp-section-maxWidth-default',
  'lazy-load': true
}, _commonProps.pointerEventsCategoryBaseDefaults);

var effects = {
  hover: ':hover'
};

var _default = Object.assign(Structure, {
  overrides: overrides,
  defaultProps: defaultProps,
  effects: effects,
  propInfo: propInfo
});

exports["default"] = _default;
