// @quarkly/components/build/cjs/prod/rawHtml/allowed-reactEls.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = isAllowed;

var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));

var allowedPlaces = ['endOfHead', 'endOfBody'];

function isAllowedStyle(reactEl) {
  return typeof reactEl.props.children === 'string' && reactEl.props.children.trim().length > 0;
}

function isAllowedScript(reactEl) {
  return typeof reactEl.props.children === 'string' && reactEl.props.children.trim().length > 0 || typeof reactEl.props.src === 'string' && reactEl.props.src.trim().length > 0;
}

var allowedTags = {
  script: isAllowedScript,
  style: isAllowedStyle,
  meta: function meta() {
    return true;
  },
  link: function link() {
    return true;
  }
};

function isAllowed(reactEl) {
  return (0, _typeof2["default"])(reactEl) === 'object' && reactEl !== null && allowedTags[reactEl.type] && allowedPlaces.includes(reactEl.props.place) && typeof reactEl.props.rawKey === 'string' && reactEl.props.rawKey.length > 0 && allowedTags[reactEl.type](reactEl);
}
