// @quarkly/components/build/cjs/prod/menu/Menu.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _atomize = _interopRequireDefault(require("@quarkly/atomize"));

var _reactRouterDom = require("react-router-dom");

var _router = require("@reach/router");

var _widgets = require("@quarkly/widgets");

var _hooks = require("../../hooks");

var _props = require("./props");

var _utils = require("./utils");

var _getQAPI = _interopRequireDefault(require("../../utils/getQAPI"));

var Ul = _atomize["default"].ul();

var Li = _atomize["default"].li();

var Item = function Item(_ref) {
  var _item$children;

  var _ref$exact = _ref.exact,
      exact = _ref$exact === void 0 ? false : _ref$exact,
      override = _ref.override,
      item = _ref.item;
  var hasSub = !!(item === null || item === void 0 ? void 0 : (_item$children = item.children) === null || _item$children === void 0 ? void 0 : _item$children.length);
  var common = {
    exact: exact,
    override: override,
    item: item
  };

  var _ref2 = (0, _getQAPI["default"])() || {},
      projectType = _ref2.projectType;

  var href = item.absoluteUrl;
  var match = null;

  if (projectType === 'gatsby') {
    match = (0, _router.useMatch)(href) || null; // eslint-disable-line
  } else {
    match = (0, _reactRouterDom.useRouteMatch)({
      path: href,
      exact: exact
    }) || null; // eslint-disable-line
  }

  var name = item.name,
      pageUrl = item.pageUrl;
  var linkProps = override('link', match && 'link-active', "link-".concat(pageUrl), {
    defaultKey: "link-".concat(pageUrl)
  });
  return /*#__PURE__*/_react["default"].createElement(Li, override('item', match && 'item-active', "item-".concat(pageUrl), {
    defaultKey: "item-".concat(pageUrl)
  }), /*#__PURE__*/_react["default"].createElement(_widgets.Link, (0, _extends2["default"])({
    href: href
  }, linkProps), linkProps.children || name), hasSub && /*#__PURE__*/_react["default"].createElement(Wrapper, (0, _extends2["default"])({
    tree: item
  }, common, override('sub', "sub-".concat(pageUrl)))));
};

var Wrapper = function Wrapper(_ref3) {
  var _tree$children;

  var override = _ref3.override,
      exact = _ref3.exact,
      tree = _ref3.tree,
      rest = (0, _objectWithoutProperties2["default"])(_ref3, ["override", "exact", "tree"]);
  var common = {
    override: override,
    exact: exact,
    tree: tree
  };
  var list = (_tree$children = tree === null || tree === void 0 ? void 0 : tree.children) !== null && _tree$children !== void 0 ? _tree$children : [];
  return /*#__PURE__*/_react["default"].createElement(Ul, rest, list.map(function (item) {
    return /*#__PURE__*/_react["default"].createElement(Item, (0, _extends2["default"])({
      key: item.id,
      item: item
    }, common));
  }));
};

var Menu = function Menu(_ref4) {
  var _ref4$rootId = _ref4.rootId,
      rootUrl = _ref4$rootId === void 0 ? '' : _ref4$rootId,
      depth = _ref4.depth,
      filterMode = _ref4.filterMode,
      origFilterPages = _ref4.filterPages,
      exact = _ref4['exact-active-match'],
      props = (0, _objectWithoutProperties2["default"])(_ref4, ["rootId", "depth", "filterMode", "filterPages", "exact-active-match"]);

  var _useOverrides = (0, _hooks.useOverrides)(props, _props.overrides, _props.defaultProps),
      override = _useOverrides.override,
      rest = _useOverrides.rest;

  var pages = (0, _getQAPI["default"])().pages || {};
  var filterPages = (origFilterPages === null || origFilterPages === void 0 ? void 0 : origFilterPages.length) > 0 ? origFilterPages.split(',') : [];

  var tree = _utils.PageTreeNode.fromPages(pages).findSubtreeByUrl(rootUrl).filterByPages(filterMode, filterPages).truncate(depth);

  return /*#__PURE__*/_react["default"].createElement(Wrapper, (0, _extends2["default"])({
    tree: tree,
    depth: depth,
    exact: exact,
    override: override,
    padding: "6px",
    margin: "0px",
    "list-style": "none"
  }, rest));
};

Object.assign(Menu, {
  title: 'Menu',
  description: {
    en: 'Use this component to add a simple menu to your website',
    ru: 'Простое меню для вашего сайта'
  },
  defaultProps: _props.defaultProps,
  overrides: _props.overrides,
  propInfo: _props.propInfo
});
var _default = Menu;
exports["default"] = _default;
